practice-app/practice-app.py

```diff
@@ -82,32 +88,98 @@ def create_book():
+        cur.execute("INSERT INTO Books(book_title, book_author, url, publication_dt, summary, uuid, uri) VALUES (?,?,?,?,?,?,?)",
+                    (Book.book_title, Book.book_author, Book.url, Book.publication_dt, Book.summary, Book.uuid, Book.uri))
     except Exception as err:
         return Response(str(err), status=403)
     # since isbn's can be a list. Insert those to a seperate table.
     cur.execute("select * from Books")
     if Book.isbn13 is not []:
-        data = cur.execute("SELECT book_id FROM Books WHERE book_title = ? AND book_author = ?", (Book.book_title,Book.book_author))
+        data = cur.execute("SELECT book_id FROM Books WHERE book_title = ? AND book_author = ?",
+                           (Book.book_title, Book.book_author))
         book_id = data.fetchone()[0]
         for isbn in list(set(Book.isbn13)):
-            cur.execute("INSERT INTO BookISBNs(book_id, isbn) VALUES (?,?)", (book_id,isbn))
-    
+            cur.execute(
+                "INSERT INTO BookISBNs(book_id, isbn) VALUES (?,?)", (book_id, isbn))
+
     con.commit()
     con.close()
-    return Response("Book added succesfully!" ,status=200)
+    return Response("Book added succesfully!", status=200)
+
+
+@app.route('/movie/', methods=['GET'])
+def get_movies():
+    if "keyword" not in request.args:
+        return Response("Please provide a keyword!", status=400)
+
+    # now we are sure keyword parameter is supplied, request rewiew the moves with the keywords from NYTimes API.
+    keyword = request.args.get("keyword").title().replace(" ", "+")
+    r = requests.get(
+        "https://api.nytimes.com/svc/movies/v2/reviews/search.json?query={}&api-key=gJkqRRyjYRV0YDiUDAEXwsa0uZLL6YLh".format(keyword))
+    r = r.json()
+
+    # now we have the movies.
+    # ------------------------------------------------
+    movies = [mapper.movie_mapper(s) for s in r["results"]]
 
+    # storing movies in database for further references.
+    con = sqlite3.connect(
+        "c:/Users/HP/Desktop/2021SpringGroup12/practice-app/sqlfiles/practice-app.db")
+    cur = con.cursor()
+    for movie in movies:
+        # -----------------------------------
+        try:
+            cur.execute("INSERT INTO Movie(display_title, mpaa_rating, critics_pick, byline, headline,summary_short, link) VALUES (?,?,?,?,?,?,?)",
+                        (movie.display_title, movie.mpaa_rating, movie.critics_pick, movie.byline, movie.headline, movie.summary_short, movie.link.url))
+        except:
+            # do nothing upon failure, this is not a critical process
+            continue
+    con.commit()
+    con.close()
 
+    return schemas.MovieResponse(movies=movies).__dict__
 
+# ---------------------------------------------------------------------------------------
 
 
+@app.route('/movie/', methods=['POST'])
```

Comment:
Its a best practice to make endpoint paths plural. I mean: /movie -> /movies

Reply:
Agreed